hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,486 @@
+        input_data: Input package containing the specific input data registers
+
+    Raises:
+        Exception: If computer is not able to establish comminucation with robot
+    """
+
+    conf = rtde_config.ConfigFile(Path(Path.cwd() / "robot_communication_file.xml"))
+    output_names, output_types = conf.get_recipe("out")
+    input_names, input_types = conf.get_recipe("in")
+
+    # Rtde communication is by default port 30004.
+    con = rtde.RTDE(host, port)
+    con.connect()
+
+    # To ensure that the application and further versions of UR controller is compatible
+    if not con.negotiate_protocol_version():
+        raise Exception(f"Protocol do not match")
+
+    if not con.send_output_setup(output_names, output_types, frequency=200):
+        raise Exception(f"Unable to configure output")
+
+    input_data = con.send_input_setup(input_names, input_types)
+
+    if not con.send_start():
+        raise Exception(f"Unable to start synchronization")
+
+    print("Communication initialization completed. \n")
+
+    return con, input_data
+
+
+def _save_zdf_and_pose(
+    save_dir: Path, image_num: int, frame: zivid.Frame, transform: np.array
+):
+    """Save data to folder
+
+    Args:
+        save_dir: Directory to save data
+        image_num: Image number
+        frame: Point cloud stored as .zdf
+        transform: 4x4 transformation matrix
+    """
+
+    frame.save(save_dir / f"img{image_num:02d}.zdf")
+
+    file_storage = cv2.FileStorage(
+        str(save_dir / f"pos{image_num:02d}.yaml"), cv2.FILE_STORAGE_WRITE
+    )
+    file_storage.write("PoseState", transform)
+    file_storage.release()
+
+
+def _generate_folder():
+    """Generate folder where dataset weill be stored
+
+    Returns:
+        location_dir: The directory to save data
+    """
+
+    location_dir = Path(
+        Path.cwd() / "datasets" / datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    )
+    if not location_dir.is_dir():
+        location_dir.mkdir(parents=True)
+
+    return location_dir
+
+
+def _capture_and_get_robot_pose(con: rtde, cam: zivid.Camera):
+    """Capture image with Zivid camera and read robot pose
+
+    Args:
+        con: Connection between computer and robot
+        cam: Zivid camera
+
+    Returns:
+        frame: Point cloud stored as .zdf
+        image_number: Image number
+        robot_pose_before: Pose of robot taken right before image was captured
+        robot_pose_after: Pose of robot right after image was captured
+    """
+
+    robot_state = con.receive()
+    image_num = robot_state.output_int_register_24
+    robot_pose_before = np.array(robot_state.actual_TCP_pose)
+    frame = cam.capture()
+    robot_pose_after = np.array(con.receive().actual_TCP_pose)
+
+    return frame, image_num, robot_pose_before, robot_pose_after
+
+
+def _set_camera_settings(cam: zivid.Camera):
+    """Set camera settings
+
+    Args:
+        cam: Zivid camera
+    """
+
+    settings = zivid.Settings()
```

Comment:
Consider doing `settings = cam.settings` to ensure that you get the defaults for the actual connected camera.

Reply:
Done